### PR TITLE
Split up builds status function.

### PR DESCRIPTION
**Pull request by oliverchang**

### CODE REVIEW

**Comment by oliverchang** [comment]
fixes #4255

File: infra/build/functions/update_build_status.py [hunk 1]
```diff
@@ -233,19 +243,58 @@ def upload_log(build_id):
...
+    return
+
+  if status_type == 'fuzzing':
+    tag = build_project.FUZZING_BUILD_TAG
+    status_filename = FUZZING_STATUS_FILENAME
+  elif status_type == 'coverage':
+    tag = build_and_run_coverage.COVERAGE_BUILD_TAG
+    status_filename = COVERAGE_STATUS_FILENAME
+  else:
+    raise RuntimeError('Invalid type ' + status_type)
```
Comment (inferno-chromium) [inline infra/build/functions/update_build_status.py:264]: 'Invalid build status type: '

File: infra/build/functions/update_build_status.py [hunk 2]
```diff
@@ -233,19 +243,58 @@ def upload_log(build_id):
 # pylint: disable=no-member
 def update_status(event, context):
   """Entry point for cloud function to update build statuses and badges."""
-  del event, context  #unused
+  del context
+
+  if 'data' in event:
+    status_type = base64.b64decode(event['data']).decode()
+  else:
+    raise RuntimeError('No data')
```
Comment (inferno-chromium) [inline infra/build/functions/update_build_status.py:251]: No build status data.

File: infra/build/functions/update_build_status.py [hunk 3]
```diff
@@ -233,19 +243,58 @@ def upload_log(build_id):
...
+
+  return result
+
+
+def update_badges():
+  """Update badges."""
+  project_build_statuses = load_status_from_gcs(FUZZING_STATUS_FILENAME)
+  coverage_build_statuses = load_status_from_gcs(COVERAGE_STATUS_FILENAME)
+
+  with concurrent.futures.ThreadPoolExecutor(max_workers=32) as executor:
```
Comment (inferno-chromium) [inline infra/build/functions/update_build_status.py:288]: Do we really need that high parallelism here - even a common global var of 8 or 10 is fine.

File: infra/build/functions/update_build_status.py [hunk 4]
```diff
@@ -162,24 +167,29 @@ def get_build_history(build_ids):
...
-      BuildsHistory.build_tag == build_tag).order('project'):
 
+  def process_project(project_build):
+    """Process a project."""
     project = get_build_history(project_build.build_ids)
     project['name'] = project_build.project
-    projects.append(project)
-    if project['history']:
-      statuses[project_build.project] = project['history'][0]['success']
+    print('Up to', project['name'])
```
Comment (inferno-chromium) [inline infra/build/functions/update_build_status.py:175]: Maybe "Processing project: "

File: infra/build/functions/update_build_status.py (hunk 2 above)
Comment (oliverchang) [reply infra/build/functions/update_build_status.py:251]: This isn't really "builds status" data, as it's referring to the generic "data" that the pub/sub trigger gives us.

File: infra/build/functions/update_build_status.py (hunk 1 above)
Comment (oliverchang) [reply infra/build/functions/update_build_status.py:264]: Done.

File: infra/build/functions/update_build_status.py (hunk 3 above)
Comment (oliverchang) [reply infra/build/functions/update_build_status.py:288]: It makes things much faster here. The reason for the lower limit for the other case is due to QPS restrictions.

File: infra/build/functions/update_build_status.py (hunk 4 above)
Comment (oliverchang) [reply infra/build/functions/update_build_status.py:175]: done.

File: infra/build/functions/update_build_status.py [hunk 5]
```diff
@@ -233,19 +243,58 @@ def upload_log(build_id):
...
+    return
+
+  if status_type == 'fuzzing':
+    tag = build_project.FUZZING_BUILD_TAG
+    status_filename = FUZZING_STATUS_FILENAME
+  elif status_type == 'coverage':
+    tag = build_and_run_coverage.COVERAGE_BUILD_TAG
+    status_filename = COVERAGE_STATUS_FILENAME
+  else:
+    raise RuntimeError('Invalid build status type ' + status_type)
```
Comment (Dor1s) [inline infra/build/functions/update_build_status.py:264]: nit: I'd personally prefer log / error messages in a format like: `'Invalid build status type:' + status_type` (a colon to separate string literal and the actual value, same for other prints / exceptions), but that's just a preference I guess and definitely isn't worth another CL to address

**Pull request merged**